lib/logstash/filters/dns.rb

```diff
@@ -142,7 +142,13 @@ def resolve(event)
       begin
         return if @failed_cache && @failed_cache[raw] # recently failed resolv, skip
         if @hit_cache
-          address = @hit_cache.getset(raw) { retriable_getaddress(raw) }
+          address = @hit_cache[raw]
+          if address.nil?
+            address = retriable_getaddress(raw)
```

Comment:
minor styling preference here:

```ruby
if address = retriable_getaddress(raw)
  @hit_cache[raw] = address
end
```

Reply:
Yeah that is better, isn't it.
Do I need to change and resubmit the PR?

Reply:
you can write an extra commit and push to your own `robcowart:concurrent_queries_with_cache`git branch, it will update this PR

Reply:
OK. Standby.